### ISSUE TITLE
Gazillions of log messages "Use of uninitialized value ..."

### ISSUE THREAD

**Issue by vpeil** [LibreCat/Catmandu-FileStore#5]
```bash
Use of uninitialized value in addition (+) at /srv/www/pub/LibreCat/local/lib/perl5/Catmandu/Store/File/Simple/Bag.pm line 97.
```

Logs, produced by LibreCat, explode.

**Comment by nicolasfranck** [comment]
this code (rule 97):
```
$bytes += $out->write($buffer);
```
assumes that the number of bytes written is returned, which is not.
I had a look at the documentation of IO::Handle:

"[..]  However, whilst the C write function returns the number of bytes written, this write function simply returns true if successful (like print). A more C-like write is syswrite"

So it returns either 1 or undef.

**Comment by phochste** [comment]
Confirmed bug

**Comment by phochste** [comment]
Fixed in 0.12

**Comment by vpeil** [comment]
thanks.
